ultravox/inference/infer.py

```diff
@@ -33,22 +33,33 @@ def __init__(
     def infer(
         self,
         sample: datasets.VoiceSample,
-        max_tokens: Optional[int] = None,
+        max_new_tokens: Optional[int] = None,
         temperature: Optional[float] = None,
+        past_key_values: Optional[Union[Tuple, transformers.cache_utils.Cache]] = None,
```

Comment:
This parameter doesn't seem quite right to me since it has no meaning in other implementations. Instead could we just keep an internal cache within `LocalInference`?

Reply:
Good suggestion. Updated to include `past_messages` and `past_key_values` within `LocalInference`